cmake/FindFreeRTOS.cmake

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+# Modified version from https://github.com/ObKo/stm32-cmake
+
+if(NOT FreeRTOS_FIND_COMPONENTS)
+    set(FreeRTOS_FIND_COMPONENTS
+        ARM_CM0 ARM_CM3 ARM_CM4F ARM_CM7 Posix
+    )
+endif()
+list(REMOVE_DUPLICATES FreeRTOS_FIND_COMPONENTS)
+
+set(FreeRTOS_HEAPS 1 2 3 4 5)
+
+if(NOT FREERTOS_PATH)
+    set(FREERTOS_PATH /opt/FreeRTOS CACHE PATH "Path to FreeRTOS")
+    message(STATUS "No FREERTOS_PATH specified using default: ${FREERTOS_PATH}")
+endif()
+
+find_path(FreeRTOS_COMMON_INCLUDE
+    NAMES FreeRTOS.h
+    PATHS "${FREERTOS_PATH}" "${FREERTOS_PATH}/FreeRTOS" 
+    PATH_SUFFIXES  "Source/include"
+    NO_DEFAULT_PATH
+)
+list(APPEND FreeRTOS_INCLUDE_DIRS "${FreeRTOS_COMMON_INCLUDE}")
+
+find_path(FreeRTOS_SOURCE_DIR
+    NAMES tasks.c
+    PATHS "${FREERTOS_PATH}" "${FREERTOS_PATH}/FreeRTOS" 
+    PATH_SUFFIXES  "Source"
+    NO_DEFAULT_PATH
+)
+if(NOT (TARGET FreeRTOS))
+    add_library(FreeRTOS INTERFACE IMPORTED)
+    target_sources(FreeRTOS INTERFACE 
+        "${FreeRTOS_SOURCE_DIR}/tasks.c"
+        "${FreeRTOS_SOURCE_DIR}/list.c"
+        "${FreeRTOS_SOURCE_DIR}/queue.c"
+    )
+    target_include_directories(FreeRTOS INTERFACE "${FreeRTOS_COMMON_INCLUDE}")
```

Comment:
nit: maybe make this `FreeRTOS::Kernel` or `FreeRTOS::FreeRTOS`

Reply:
for now I want to keep it in sync with https://github.com/ObKo/stm32-cmake version